_glaredb/python.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: default
+title: Python bindings
+nav_order: 2
+---
+
+# Use GlareDB with Python
+
+GlareDB can be used in Python environments along with [Pandas] and [Polars] to
+work with distributed data.
+
+## Install
+
+The `glaredb` module is available on PyPI <https://pypi.org/project/glaredb/>.
+To install the module:
+
+```console
+pip install glaredb
+```
+
+## Usage
+
+{: .important}
+
+> GlareDB is run as a local process in Python environments. Connections to remote
+> instances of GlareDB will be supported in a future version.
+
+- import the `glaredb` module
+- `connect` and then either `execute` eagerly or lazily with `sql`
+
+### API
+
+- `module glaredb`
+  - `connect([data_dir], [spill_path]): LocalSession`
+    - `data_dir` optional - if provided, data will be stored at this path,
+      otherwise all data is in memory.
+    - `spill_path` optional - a path where glaredb can create temporary files.
+      If not provided, a random temporary directory is used.
+- `class LocalSession`
+  - `sql(query): PyLogicalPlan`
+    - Lazily evaluates the `query` string
+  - `execute(query): PyExecutionResult`
+    - Eagerly evaluates the `query` string
+- `class PyLogicalPlan`
+  - `to_arrow()`
+  - `to_pandas()`
+  - `to_polars()`
+- `class PyExecutionResult`
+  - `to_arrow()`
+  - `to_pandas()`
+  - `to_polars()`
```

Comment:
@universalmind303 is there an expected format for python api docs? Anything we should try to follow?

Reply:
I think overall this looks fine. Eventually we could use [sphinx](https://www.sphinx-doc.org/en/master/) or similar to autogenerate these definitions.